lib/tokenable.rb

```diff
@@ -0,0 +1,11 @@
+module Tokenable
+  extend ActiveSupport::Concern
+
+  included do
+    before_create :generate_token
```

Comment:
We have a security flaw in our solution. Generating token at the beginning when we create user object means that anyone who would 'steal' our data or db, could access users account and then change pw. So its like unencrypted password. Better way would be to only generate it on request, or even better to encrypt it too. But thats beyond the topic for now.